### DISCUSSION TITLE
Introductions

### DISCUSSION THREAD

**Discussion by fkorotkov** [cirruslabs/tart#857] [Show and tell]
Please feel free to share how you use Tart within your organization or project. 🚀 Bragging is allowed in this thread. 😅 

It will help the team to prioritize work and others get inspired.

**Comment by pauldsmyth** [comment]
Well, it looks like nobody wants to show off their projects so I'll jump in.
I've got a sizeable, noisy and power hungry homelab. Mostly Dell R220s, a generic dual xeon NAS and a Dell R420.
I run the usual slew of Home automation/IoT applications. Truenas on my NAS, Home Assistant, Plex etc. All dockerised. 
My plan is to move from docker to k3s and ArgoCD. I've got some Intel Mac Minis and want to also run a M4 Mac Mini as the heart of it for the heaviest lifting. 

I looked at VMWare Fusion, UTM, Colima and Lima. The first two have the advantage of being able to mount USB devices but I found VMWare very flaky. UTM works well but I wanted something more native. 

I looked at Colima but that doesn't support multinode k3s so I was looking at going upstream to Lima. This looked promising but the networking seemed to be not very well explained. I was looking for networking examples and discovered Tart from a discussion thread and to be honest, I was up and running so quickly that I was pretty much decided this was the way to go.

My network set up is 2 bridged connections. The first is the lan and the second is a storage connection over a vlan. With a little trial and error and having to manually install vlan, it was running very quickly. The next step is to run for a couple more days then install k3s and add it into my existing cluster. 

Potential future enhancements that would take this to the next level:

1. Ability to start/stop on boot/shutdown cleanly. Particularly in the event of a power outage so that I can monitor the UPS and shutdown cleanly.
2. Ability to slipstream files like netplan yaml files into install cloud-init style. 
3. Ability to attach a USB device (TPU, Bluetooth, ZigBee dongle etc)

Very impressed with this project. I only found it last night and I'm up and running. Fantastic job

**Reply by edigaryev** [reply]
>Ability to slipstream files like netplan yaml files into install cloud-init style.

I think it's already possible with Packer, please check out [our Cloud Init template for Linux](https://github.com/cirruslabs/linux-image-templates/blob/27e05e72440cb4984d61923157047cc2265984de/cloud-init.pkr.hcl#L17).

**Reply by pauldsmyth** [reply]
Thanks, fantastic. This project has really impressed me

**Comment by dweekly** [comment]
Hey folks! I'm using Tart at my tiny, brand-new startup Network Weather to build self-hosted testing and CI/CD infrastructure for a desktop macOS app that I'm building to help end-users self-triage network connectivity issues. I especially appreciate the bridged networking mode and ease of running GitHub Actions runners. It's great to be able to easily test the app across macOS 12, 13, 14, and 15!

**Comment by webcoyote** [comment]
Tart is amazing! I used it to create [ClodPod](https://github.com/webcoyote/clodpod), an open-source (Apache License) script which builds an OSX virtual machine for running Claude Code. It configures the VM to have access to only project directories instead of your whole computer so you can safely run `claude --dangerously-skip-permissions`.

**Reply by fkorotkov** [reply]
Nice one! I guess great minds think alike! We also experimented a similar project called [Chamber](https://github.com/cirruslabs/chamber).

**Comment by viewyonder** [comment]
Hello all, I just discovered Tart (recommended by Claude Code!) as the answer to clean testing of a Claude Code plugin I'm doing called Coherence (for Claude). The plugin install/update/uninstall/cleanup is a bit wonky-donkey in Claude-- look at these burned fingers and singed sideburns --so, as an old skool VMware guy, with an aversion to Docker, I wanted an easy Mac-native way to set up a simple clean env, install claude, add a repo, then install coherence, do stuff -- document the lot -- then uninstall it, still documenting, and roll back to a clean slate.

New to Tart. Will write up my experience on the viewyonder/coherence when I'm done. Happy to be hear and learn from you all.
Steve, from Yorkshire, England.
PS. Why "Tart"?

**Comment by hooleahn** [comment]
I've been working on a macOS app that uses tart, mainly focused on Mac Admins. I plan to release it soon, open-source, free as in beer and all of that. Are there any licensing concerns I should take into account?

And also, is there an appetite for those kinds of apps?
